Fabric.Authorization.API/Modules/UsersModule.cs

```diff
@@ -68,6 +79,28 @@ public class UsersModule : FabricModule<User>
             Get("/{identityProvider}/{subjectId}/groups",
                 async _ => await GetUserGroups().ConfigureAwait(false), null,
                 "GetUserGroups");
+
+            Get("/{identityProvider/{subjectId}/roles",
+                async _ => await GetUserRoles().ConfigureAwait(false), null,
+                "GetUserRoles");
+
+            Post("/{identityProvider}/{subjectId}/roles",
+                async param => await AddRolesToUser(param).ConfigureAwait(false), null,
+                "AddRolesToUser");
+        }
+
+        private async Task<dynamic> GetUser(dynamic param)
+        {
+            CheckReadAccess();
+            try
+            {

```

Comment:
Verify `subjectId` and `identityProvider` are not null?

Reply:
I don't think they can be null, since it wouldn't be the route without those params, and the NotFoundException should catch this one anyway.